play-media-native/screens/CreateAthlete/CreateAthlete.tsx

```diff
@@ -0,0 +1,79 @@
+import { Screen } from "../../features/Screen/Screen";
+import { Stepper } from "../../components/Stepper/Stepper";
+import { useCallback, useMemo, useState } from "react";
+import { General } from "./General";
+import { BottomActions } from "../../components/BottomActions/BottomActions";
+import { Button } from "react-native-paper";
+import { styles } from "../../theme/styles";
+import { athleteStyles } from "./styles";
+import { useQuery } from "react-query";
+import { getAllSports } from "../../api/queries/getSports";
+import { LoadingScreen } from "../../features/LoadingScreen/LoadingScreen";
+
+export const CreateAthleteScreen = ({ navigation }) => {
+  let [step, setStep] = useState(0);
+  const [sports, setSports] = useState([]);
+
+  const steps = ["General", "Content", "References"];
+
+  const onStepPress = useCallback((index: number) => {
+    setStep(index);
+  }, []);
+
+  const displayedScreen = useMemo(() => {
+    if (step === 0) {
+      return <General sports={sports} />;
+    }
+
+  }, [step, sports]);
+
+  const { data, isFetching } = useQuery("sports", () => getAllSports(), {
+    onSuccess: (data) => setSports(data),
+  });
+
+  if (isFetching) {
+    return <LoadingScreen />;
+  }
+
+  const handleDiscardBtn = () => {
+    navigation.goBack();
+  };
+  const handleNextBtn = () => {
+    onStepPress(++step);
+  };
+  const handleSubmitBtn = () => {
+    // TODO Submit API request
+  };
+
+  return (
+    <Screen>
+      <Stepper
+        labels={steps}
+        onPress={onStepPress}
+        stepIndex={step}
+        steps={steps}
+      />
+      {displayedScreen}
+      <BottomActions style={athleteStyles.actionBtns}>
+        <Button
+          mode="outlined"
+          style={styles.button}
+          labelStyle={styles.buttonLabel}
+          onPress={() => handleDiscardBtn()}
```

Comment:
just give it onPress={handleDiscardBtn}. It would make sense to have it like this if an argument was passed in , example () => handleSomething(id), but biw it;s just extra JS work that is not needed

Reply:
True, it just looks prettier 😄. I'll update

Reply:
It does yes :). Just keep in mind that every time any state or prop changes here, the component will re-create any non-memoized function. Especially if handleDiscardBtn is not wrapped in useCallback, then on every re-render it will re-create 2 functions, one is handleDiscardBtn and the other is the anonymous one on this line.